lte/gateway/python/magma/pipelined/app/conntrack.py

```diff
@@ -0,0 +1,231 @@
+                            match, actions,
+                            priority=flows.DEFAULT_PRIORITY)
+
+        # Match tcp terminations (fin)
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                           ip_proto=IPPROTO_TCP,
+                           tcp_flags=(0x1,0x1),
+                           ct_state=(self.CT_EST | self.CT_TRK,
+                                     self.CT_EST | self.CT_TRK))
+        actions = [parser.NXActionCT(
+            flags=0x0,
+            zone_src=None,
+            zone_ofs_nbits=0,
+            recirc_table=self.connection_event_table,
+            alg=0,
+            actions=[]
+        )]
+        flows.add_drop_flow(datapath, self.conntrack_scratch,
+                            match, actions,
+                            priority=flows.DEFAULT_PRIORITY)
+        # match tcp fin
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                           ip_proto=IPPROTO_TCP,
+                           tcp_flags=(0x1, 0x1),
+                           ct_state=(self.CT_TRK | self.CT_INV, self.CT_TRK | self.CT_INV))
+        # flags 0 is nothing, 1 is commit
+        actions = [parser.NXActionCT(
+            flags=0x0,
+            zone_src=None,
+            zone_ofs_nbits=0,
+            recirc_table=self.connection_event_table,
+            alg=0,
+            actions=[]
+        )]
+        flows.add_drop_flow(datapath, self.conntrack_scratch,
+                            match, actions,
+                            priority=flows.DEFAULT_PRIORITY)
+
+        # match tcp rst
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                           ip_proto=IPPROTO_TCP,
+                           tcp_flags=(0x4, 0x4),
+                           ct_state=(self.CT_EST | self.CT_TRK, self.CT_EST | self.CT_TRK))
+        # flags 0 is nothing, 1 is commit
+        actions = [parser.NXActionCT(
+            flags=0x0,
+            zone_src=None,
+            zone_ofs_nbits=0,
+            recirc_table=self.connection_event_table,
+            alg=0,
+            actions=[]
+        )]
+        flows.add_drop_flow(datapath, self.conntrack_scratch,
+                            match, actions,
+                            priority=flows.DEFAULT_PRIORITY)
+
+        inbound_match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                                   direction=Direction.IN)
+        outbound_match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                                    direction=Direction.OUT)
+        flows.add_resubmit_next_service_flow(
+            datapath, self.tbl_num, inbound_match, [],
+            priority=flows.MINIMUM_PRIORITY,
+            resubmit_table=self.next_table)
+        flows.add_resubmit_next_service_flow(
+            datapath, self.tbl_num, outbound_match, [],
+            priority=flows.MINIMUM_PRIORITY,
+            resubmit_table=self.next_table)
+
```

Comment:
I dont see need to match on direction, given that action for both flows is same.

Reply:
Yeah for some reason these are our default flows for a lot of controllers. I'm actually not sure why it was done this way initially. I've always been to lazy but I think I might finally refactor all controllers to use 1 default rule instead of 2.